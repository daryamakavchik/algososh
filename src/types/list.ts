import { ElementStates } from "./element-states";

export interface ILinkedList<T> {
  append: (element: T) => void;
  prepend: (element: T) => void;
  deleteHead: () => void;
  deleteTail: () => void;
  addByIndex: (element: T, index: number) => void;
  removeByIndex: (index: number) => void;
}

export type TListItem = {
  value: string;
  color: ElementStates;
};