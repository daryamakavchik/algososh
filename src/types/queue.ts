import { ElementStates } from "./element-states";
export interface IQueue<T> {
    enqueue: (item: T) => void;
    dequeue: () => void;
    peek: () => T | null;
  }

export type TQueueItem = {
    value?: string;
    color: ElementStates;
    head?: string;
  };