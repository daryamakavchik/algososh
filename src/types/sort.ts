import { ElementStates } from "./element-states";
export type TArray = {
    value: number;
    color: ElementStates;
  };