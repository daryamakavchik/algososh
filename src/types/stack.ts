import { ElementStates } from "./element-states";

export type TStackItem = {
  value: string;
  color: ElementStates;
};

export interface IStack<T> {
  push: (item: T) => void;
  pop: () => void;
  peek: () => T | null;
  getSize: () => number;
}
