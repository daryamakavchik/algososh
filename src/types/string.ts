import { ElementStates } from "./element-states";
export type TArray = {
    value: string;
    color: ElementStates;
  };
  