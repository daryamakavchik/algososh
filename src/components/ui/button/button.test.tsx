import React from "react";
import { Button } from "./button";
import { render, screen, fireEvent } from "@testing-library/react";
import { Direction } from "../../../types/direction";
import * as renderer from "react-test-renderer";

describe("Кнопка рендерится без ошибок", () => {
  it("Кнопка с текстом", () => {
    const tree = renderer.create(<Button text="test" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("Кнопка без текста", () => {
    const tree = renderer.create(<Button />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("Деактивированная кнопка", () => {
    const tree = renderer.create(<Button disabled />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("Кнопка с загрузкой", () => {
    const tree = renderer.create(<Button isLoader={true} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("Кнопка сортировки по возрастанию", () => {
    const tree = renderer.create(<Button text="По возрастанию"
    sorting={Direction.Ascending} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("Кнопка сортировки по убыванию", () => {
    const tree = renderer.create(<Button text="По возрастанию"
    sorting={Direction.Descending} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("Кнопка меньшей ширины", () => {
    const tree = renderer.create(<Button linkedList="small" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("Кнопка большей ширины", () => {
    const tree = renderer.create(<Button linkedList="big" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("Кнопка работает при нажатии", () => {
    const callBack = jest.fn();
    render(<Button onClick={callBack} />);
    fireEvent.click(screen.getByRole("button"));
    expect(callBack).toHaveBeenCalled();
  });
});
