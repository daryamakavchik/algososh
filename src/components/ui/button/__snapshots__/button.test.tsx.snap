// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Кнопка рендерится без ошибок Деактивированная кнопка 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  disabled={true}
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[`Кнопка рендерится без ошибок Кнопка без текста 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[`Кнопка рендерится без ошибок Кнопка с загрузкой 1`] = `
<button
  className="text text_type_button text_color_primary button undefined loader "
  disabled={true}
  type="button"
>
  <img
    alt="Загрузка."
    className="loader_icon"
    src="loader.svg"
  />
</button>
`;

exports[`Кнопка рендерится без ошибок Кнопка с текстом 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  >
    test
  </p>
</button>
`;
